solarforecastarbiter/validation/tasks.py

```diff
@@ -652,9 +580,144 @@ def apply_validation(observation, observation_values):
     if data.empty:
         return data
     if (
-            (data.index[-1] - data.index[0]) > pd.Timedelta('1d') and
+            (data.index[-1] - data.index[0]) >= pd.Timedelta('1d') and
             (len(data['value'].dropna()) > 10)
     ):
         return apply_daily_validation(observation, data)
     else:
         return apply_immediate_validation(observation, data)
+
+
+def _group_continuous_week_post(session, observation, observation_values):
+    # observation_values expected to be sorted
+    # observation values already have uneven frequency checked
+    gid = quality_mapping.check_if_series_flagged(
+        observation_values['quality_flag'], 'UNEVEN FREQUENCY').cumsum()
+    # make series of week + year integers to further
+    # split data to post at most one week at a time
+    # ~10,000 pts of 1min data
+    week_int = (gid.index.week + gid.index.year).values
+    # combine the continuous groups with groups of weeks
+    # gid is unique for each group since week_int and cumsum
+    # increase monotonically and are positive
+    gid += week_int
+    observation_values['gid'] = gid
+    for _, group in observation_values.groupby('gid'):
+        session.post_observation_values(observation.observation_id,
+                                        group[['value', 'quality_flag']],
+                                        params='donotvalidate')
+
+
+def _validate_post(session, observation, start, end):
+    logger.info('Validating data for %s from %s to %s',
+                observation.name, start, end)
+    observation_values = session.get_observation_values(
+        observation.observation_id, start, end)
+    validated = apply_validation(observation, observation_values)
+    return _group_continuous_week_post(
+        session, observation, validated)
+
+
+def _find_unvalidated_time_ranges(session, observation, min_start, max_end):
+    """Find the time ranges where the observation data needs to have
+    daily validation applied. Extend to next day midnight so daily
+    validation can be applied even since it requires >= 1 day of data
+    """
+    tz = observation.site.timezone
+    dates = np.array(sorted(
+        session.get_observation_values_not_flagged(
+            observation_id=observation.observation_id,
+            start=min_start,
+            end=max_end,
+            flag=(
+                quality_mapping.DAILY_VALIDATION_FLAG |
+                quality_mapping.LATEST_VERSION_FLAG
+            ),
+            timezone=tz)
+        ))
+    if len(dates) == 0:
+        return
+
+    breaks = np.diff(dates).astype('timedelta64[D]') - 1
```

Comment:
I don't see the subtraction behavior clearly documented on the first numpy page I googled, so add a comment. I assume it's subtracting a day but my mind first goes to pandas nanoseconds and then to posix seconds.